Tracks/michael-java/searchSubstring.java

```diff
@@ -0,0 +1,81 @@
+package searchSubstring;
+
+public class searchSubstring {
+
+    static String words = "... мама -папа, (дед+баба) ели к_а_ш_у!";
+    static String separators = "., !-:+";
+
+    public static void main(String[] args) {
+
+        int countWords = 0;
+
+        countWords = countWords ( words, separators );
+        printResult(countWords);
+
+        System.out.println (testCode() ? "тест пройден" : "тест не пройден");
+    }
+
+    static int countWords( String testWords, String testSeparators ){
+
+        boolean isWasLetter = false;   // состояние - буква или нет
+        int counterWords = 0;          // счетчик слов
+
+        for (int index = 0; index < testWords.length(); index++){       // перебор элементов тестируемой строки на буква/не буква
+
+            if ( isLetter( index, testWords, testSeparators ) ){        // элемент строки - буква

```

Comment:
Code style:
for (int index = 0; index < testWords.length(); index++){

if ( isLetter( index, testWords, testSeparators ) ){